### PR TITLE
account for jsonUrl being null in type signature

### PR DESCRIPTION
**Pull request by spacey-sooty**

### CODE REVIEW

**Comment by ThadHouse** [comment]
Does this actually work? Not sure if gson will actually parse this correctly.

**Comment by spacey-sooty** [comment]
I was concerned about that as well, I'll mock up a test for it later

**Comment by spacey-sooty** [comment]
This doesn't work, I've opened an issue https://github.com/google/gson/issues/2644

File: src/main/java/edu/wpi/first/nativeutils/vendordeps/OptionalTypeAdapter.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
+            Class<T> rawType = (Class<T>) type.getRawType().getClass();
+            if (rawType != Optional.class) {
+                return null;
+            }
+            final ParameterizedType parameterizedType = (ParameterizedType) type.getType();
+            final Type actualType = parameterizedType.getActualTypeArguments()[0];
+            final TypeAdapter<?> adapter = gson.getAdapter(TypeToken.get(actualType));
+            return new OptionalTypeAdapter(adapter);
+        }
```
Comment (spacey-sooty) [inline src/main/java/edu/wpi/first/nativeutils/vendordeps/OptionalTypeAdapter.java:30]: This should allow us to parse Optional correctly but has a few warnings to do with unchecked casts

Comment (ThadHouse) [reply src/main/java/edu/wpi/first/nativeutils/vendordeps/OptionalTypeAdapter.java:30]: I don’t think that is worth adding, for something we don’t really need anyway. Empty string is just fine and we can specify that.

Comment (spacey-sooty) [reply src/main/java/edu/wpi/first/nativeutils/vendordeps/OptionalTypeAdapter.java:30]: I'll open a new pr to default it to an empty string and close this one

**Pull request closed without merging**